src/storage/schema.rs

```diff
@@ -0,0 +1,265 @@
+use std::fmt;
+
+use crate::errors::*;
+use crate::storage::Transaction;
+
+pub const LATEST_SCHEMA_VERSION: SchemaVersion = SchemaVersion {
+    major: 0,
+    minor: 7,
+    patch: 0,
+    revision: 1,
+};
+
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub struct SchemaVersion {
+    major: u32,
+    minor: u32,
+    patch: u32,
+    revision: u32,
```

Comment:
In Go the revision was separate from the other three. I can't think of any benefit of the split so I kept them all at the same level.